edit-post/assets/stylesheets/main.scss

```diff
@@ -168,6 +168,32 @@ body.gutenberg-editor-page {
 			outline-offset: 0;
 		}
 	}
+
+	input[type=checkbox] {
```

Comment:
Should we move these styles to the "Checkbox control" or do we want them to apply globally to checkboxes?

 - Moving to CheckboxControl means, it works outside Gutenberg, outside WordPress, whenever you use the component
 - It also means that if you use a checkbox with the `CheckboxControl` component, it won't have the styles. 

So each option has pros/cons.

Reply:
Shoot, I thought I commented on this but didn't hit submit. Was gonna say the same thing, I don't think this should be a global checkbox style. We should scope it to our component and just make sure we're using the `CheckboxControl`.

Reply:
> If there's a JavaScripty future for the rest of WordPress

I think there is 😄

-----

I think we should try to contain our styles and apply them only to our components. We should simply get more things using our components, but the benefits of component-based development is that we can scope styles to an easily-reusable component and then just use it everywhere.

I'd really prefer we scope the styles to this component.

Reply:
@tofumatt I appreciate your passion for code cleanliness. I share it. My favorite t-shirt is one my friend @MichaelArestad wore a few years ago, that spells "Hella Clean Code" with sparkles around it. 

In this case, however, if I scope the checkbox component to be _just scoped to the component_, the categories aren't styled:

![screen shot 2018-08-07 at 11 22 52](https://user-images.githubusercontent.com/1204802/43767076-6e9c6900-9a34-11e8-9c4c-530d0775dc27.png)

I'm guessing that can be fixed by actually using the checkbox control as part of the categories component, and that this in turn would be the right thing to do. But you'll notice that <a href="https://github.com/WordPress/gutenberg/blob/a8f5319995e4aa8c281390e75cb2dd365c2e087d/edit-post/assets/stylesheets/main.scss#L130">the input styles</a> starting on this line are also located in this very same file. Should they also be moved to components? And in turn, which input styles will then break if we do this?

I think an argument can be made that the these form widgets, `input`, `select`, checkboxes, etc. are sort of base ingredients that should be styled in a global stylesheet, as opposed to on a scoped component basis. Or, should each of those HTML elements be a component on its own? And if so, is that a separate PR?

I'm not trying to be difficult — truly my biggest goal here is to simply make sure that all the textfields and checkboxes inside Gutenberg look consistent, and longer term that those same styles can be removed from Gutenberg because they are now default in WordPress as a whole (hence the need for the upstream tickets).

What's the best path forward?

Reply:
Hmmm... okay. So I'd say that what we really want is:

* styles are *always scoped to a component
* we use a component for everything (even making our own `checkbox` component so it has styles)

That said, you're right for now: it's too much of a pain to do that and it's out of scope.

Your PR is an improvement. Let's not let me make perfect the enemy of great! 😄